### PR TITLE
Components chapter: wording, grammar, formatting fixes

### PR DESCRIPTION
**Pull request by misicnenad**
### Checklist

- [x] **CI Verifier:** Run `./scripts/cairo_programs_verifier.sh` successfully
- [x] **Contract Tests:** Added tests to cover the changes

### CODE REVIEW

File: src/components/dependencies.md [hunk 1]
```diff
@@ -58,7 +58,4 @@ This will allow to do `switchable::ComponentState<TContractState>` -> `TContract
 {{#rustdoc_include ../../listings/applications/components_dependencies/src/countable_internal_dep_switch.cairo:contract}}
 ```
 
-The contract remains the same that the previous example, but the implementation of the `Countable` component is different:
-```rust
-{{#rustdoc_include ../../listings/applications/components_dependencies/src/contract_countable_switchable_internal.cairo:contract}}
```
Comment (misicnenad) [inline src/components/dependencies.md:63]: Redundant code, the explanation is clear enough

File: src/components/how_to.md [hunk 2]
```diff
@@ -45,4 +47,4 @@ The following contract incorporates the `Switchable` component:
 
 ## Deep dive into components
 
-You can find more in-depth information about components in the [Cairo book - Components](https://book.cairo-lang.org/ch99-01-05-00-components.html).
+You can find more in-depth information about components in [The Cairo book - Components](https://book.cairo-lang.org/ch16-02-00-composability-and-components.html).
```
Comment (misicnenad) [inline src/components/how_to.md:50]: Article "The" is part of book's name

Comment (misicnenad) [inline src/components/how_to.md:50]: Fixed link

**Review by julio4 — APPROVED** [review]
Good catch again, thank you so much!

**Pull request merged**